src/extension.ts

```diff
@@ -56,6 +58,21 @@ export function deactivate(context: vscode.ExtensionContext) {
 	}
 }
 
+async function createAndOpenADM() {
+	const selectedWorkspaceFolder: vscode.WorkspaceFolder | undefined = await vscode.window.showWorkspaceFolderPick( {placeHolder: 'Please select the workspace folder in which the new file will be created.'} );
+	if (selectedWorkspaceFolder) {
+		await vscode.window.showInputBox( {placeHolder: "Enter the name of the new AtlasMap file"})
+			.then( async (fileName: string) => {
+				var file: string = `${selectedWorkspaceFolder.uri.fsPath}/${fileName}`;
+				if (!file.toLowerCase().endsWith('.adm')) {
+					file += '.adm';
+				}
+				await vscode.workspace.fs.writeFile(vscode.Uri.file(file), Buffer.from(''));
+				await vscode.commands.executeCommand('vscode.open', vscode.Uri.file(file));
+			});

```

Comment:
using the wait notation everywhere will bemore readable, somethign like:
```suggestion

		const fileName = await vscode.window.showInputBox( {placeHolder: "Enter the name of the new AtlasMap file"});
		var file: string = `${selectedWorkspaceFolder.uri.fsPath}/${fileName}`;
		if (!file.toLowerCase().endsWith('.adm')) {
			file += '.adm';
		}
		await vscode.workspace.fs.writeFile(vscode.Uri.file(file), Buffer.from(''));
		await vscode.commands.executeCommand('vscode.open', vscode.Uri.file(file));

```